internal/metrics/metrics.go

```diff
@@ -179,12 +217,169 @@ func GetPDBCreatedByUsLabel(annotations map[string]string) string {
+	})
+
+	// Categorize pods as unhealthy
+	var unhealthyPods []corev1.Pod
+	for _, pod := range podList.Items {
+		if !isPodHealthy(&pod) {
+			unhealthyPods = append(unhealthyPods, pod)
+		}
+	}
+
+	// If all pods are healthy
+	if len(unhealthyPods) == 0 {
+		return AllHealthyPattern, nil
+	}
+
+	// Analyze failure pattern (we know we have >= 2 pods at this point)
+	totalPods := len(podList.Items)
+	totalUnhealthyPods := len(unhealthyPods)
+
+	// Split pods into oldest half and newest half
+	oldestHalf := totalPods / 2
+
+	oldestFailureCount := 0
+	newestFailureCount := 0
+
+	for i, pod := range podList.Items {
+		if !isPodHealthy(&pod) {
+			if i < oldestHalf {
+				oldestFailureCount++
+			} else {
+				newestFailureCount++
+			}
+		}
+	}
+
+	// Follow majority logic
+	// If more than 66% of failures are in oldest half -> oldest failing
+	// If more than 66% of failures are in newest half -> newest failing
+	// Otherwise this is a random failing pattern
+	oldestFailureRatio := float64(oldestFailureCount) / float64(totalUnhealthyPods)
+	newestFailureRatio := float64(newestFailureCount) / float64(totalUnhealthyPods)
+
+	if oldestFailureRatio >= 0.66 {
+		return OldestFailingPattern, nil
+	} else if newestFailureRatio >= 0.66 {
+		return NewestFailingPattern, nil
+	} else {
+		return RandomFailingPattern, nil
+	}
+}
+
+// isPodHealthy determines if a pod is healthy based on its conditions
+func isPodHealthy(pod *corev1.Pod) bool {
+	// Check if pod is running
+	if pod.Status.Phase != corev1.PodRunning {
+		return false
+	}
+
+	// Check readiness condition
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodReady {
+			return condition.Status == corev1.ConditionTrue
+		}
+	}
+
+	return false
+}
+
+// UpdatePodAgeFailureMetrics updates the metrics for pod age failure patterns
+func UpdatePodAgeFailureMetrics(ctx context.Context, c client.Client, pdb *policyv1.PodDisruptionBudget) {
+	pattern, err := analyzePodAgeFailurePattern(ctx, c, pdb)
+	if err != nil {
+		// Log error but don't fail the reconcile
+		return
+	}
+
+	// Reset all pattern metrics for this PDB
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, OldestFailingPattern).Set(0)
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, NewestFailingPattern).Set(0)
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, RandomFailingPattern).Set(0)
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, AllHealthyPattern).Set(0)
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, AbandonedPDBPattern).Set(0)
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, UnknownPattern).Set(0)
+
+	// Set the current pattern
+	PodAgeFailurePatternGauge.WithLabelValues(pdb.Namespace, pdb.Name, pattern).Set(1)
+
+	// Update scaling opportunity counter with signal
+	signal := GetScalingSignal(ctx, c, pdb)
+	ScalingOpportunityCounter.WithLabelValues(
```

Comment:
it seems wierd to increment the opportunity counter here. If we've handled all the evictions at 
	if EvictionAutoScaler.Spec.LastEviction == EvictionAutoScaler.Status.LastEviction {
	
	Why are we saying theres anotehr scaling oppurtunt? 
	
	The guages are fine but they also only get updated when an eviction starts and is then handled so might want to clarify that about them.